### DISCUSSION TITLE
Quelle signification de l'idImmeuble du relatedEntity pour un ajout de nouveau Bâtiment à  une adresse existante

### DISCUSSION THREAD

**Discussion by ericjacq92** [before-interop/anomalieAdresse#59] [General]
Par définition l'ajout d'un nouveau bâtiment implique qu'on ne connait pas encore son id d'Immeuble.
A quoi correspond alors le relatedEntity à valoriser obligatoirement dans ce cas d'AnomalieAdresseNewBuilding : est ce bien l'id d'un immeuble voisin déjà positionné à cette même adresse ?

**Comment by ggrebert** [comment]
Hello @ericjacq92 ,

C'est un champ qui doit être rempli par l'OI lors de la création de l'immeuble.
C'est pour cela que c'est un tableau qui ne peut contenir que 1 seul élément.
`id` représentant la référence de l'immeuble.

Ce système permet à l'OC de connaitre facilement la ref créé et de valider si celle-ci est bien dans le fichier IPE.

**Reply by ericjacq92** [reply]
C'est l'OC qui déclare l'anomalie AnomalieAdresseNewBuilding.
Quand on regarde l'opération POST sur AnomalieAdresseNewBuilding, le modèle impose de fournir un bloc relatedEntity avec 1 élément obligatoire, et sur ce bloc typé building de passer un id obligatoire.
Cela veut donc dire que l'OC dés la création de l'AnomalieAdresseNewBuilding doit passer un id d'immeuble obligatoirement.
Est ce une coquille du swagger du coup ?

**Reply by ggrebert** [reply]
Il faut rendre ce champ 'read only'.
Je prépare une merge request

**Comment by ggrebert** [comment]
Je reviens sur mon message précédent après une analyse plus complète.

*AnomalieAdresseNewBuilding* permet de rajouter un nouveau bâtiment à une adresse existante dans l'IPE.

Le champs `relatedEntity.id` correspond à cette adresse déjà existante et doit donc être fourni par l'OC et est obligatoire.

J'avais confondue avec le champ `building.id` qui lui est la ref IPE nouvellement créé.

**Reply by ericjacq92** [reply]
En phase sur la compréhension et de fait : le champs relatedEntity.id  est donc à valoriser avec un code immeuble déjà présent à cette même adresse dans l'IPE, ce qui correspond donc à un immeuble voisin du nouvel immeuble objet de la demande.
